api/test/wfl/unit/references_test.clj

```diff
@@ -0,0 +1,22 @@
+(ns wfl.unit.references-test
+  (:require [clojure.string :as str]
+            [clojure.test :refer [deftest testing is]]
+            [wfl.references :as references]))
+
+(deftest prefix-handling
+  (testing "reference_fasta function"
+    (testing "supplies a bucket prefix if no arg"
+      (run! #(is (str/starts-with? % "gs://"))
+            (vals (references/reference_fasta))))
+    (testing "uses a given prefix"
+      (let [prefix "foo"]
+        (run! #(is (str/starts-with? % prefix))
+              (vals (references/reference_fasta prefix))))))
+  (testing "hg38-genome-references function"
+    (testing "supplies a reference_fasta default if nil"
+      (run! #(is (str/starts-with? % "gs://"))
+            (vals (:reference_fasta (references/hg38-genome-references nil)))))
```

Comment:
Appreciate that you've taken the time to lick down behaviour in tests.
For my money, passing `nil' is never really the right thing to do as it doesn't really capture what the programmer rmeans. They don't mean a nil thing, they mean give me the default.

Reply:
Thanks, and I agree. I did it this way because XX accesses `(reference_fasta)` to get the default, while WGS uses `(make-references nil)` which calls `(hg38-genome-references nil)` which now calls `(reference_fasta)`. If I were to remove nil further up the chain on the WGS side, it would make it awkward to have arg and no-arg versions of all of those functions. I think the code is at least readable as is and the testing should form a good set of guide rails if other people venture here in the future.